### PR TITLE
[2324 2322](Feature) Added SwiftFormat and Pre-commit 

### PR DESCRIPTION
**Pull request by guilhermepratac**
# Pull Request

## Ticket or Issue link
2324 2322

## Description

- Create Makefile to get easy installation of library
- Create Mintfile*
- Create pre commit using SwiftFormat

*A **Mintfile** is a configuration file used by Mint, a package manager for Swift tools. It lists GitHub repositories and versions of Swift packages to install. Running mint bootstrap installs the specified tools from the Mintfile.
## Checklist
- [] I have tested my changes creating a local version (More info in README file).
- [] I have added tests that prove my solution is effective and works
- [] New and existing unit tests pass locally with my changes.
- [] Verify that you are merged with the latest in develop.
- [] I have run `swiftlint` and fixed any possible issues of my code.
- [] I have tested my changes with configuration changes such as rotation, no conection, framework error handling
- [] I have tested the accessibility of the changes and added them to the screenshots.
- [] I have added a tag to the pull request that contains the product this pull request is related to.

### CODE REVIEW

**Review by rodrigmatrix — APPROVED** [review]

File: .swiftlint.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+disabled_rules:
```
Comment (icaldana) [inline .swiftlint.yml]: Essas regras foram definidas baseadas em algum critério, por exemplo o padrão  de mercado?

File: scripts/swift-format-mp.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
```
Comment (icaldana) [inline scripts/swift-format-mp.sh:1]: A ideia seria rodar esse script como parte da validação da PR?

**Review by icaldana — APPROVED** [review]

File: scripts/swift-format-mp.sh (hunk 2 above)
Comment (guilhermepratac) [reply scripts/swift-format-mp.sh:1]: Para instalar o projeto, precisará executar make setup, que instalará as dependências necessárias e configurará o script no pre-commit

File: .swiftlint.yml (hunk 1 above)
Comment (guilhermepratac) [reply .swiftlint.yml]: É um arquivo que vem como padrão, retirei disabled_rules por hora, caso precise retirar algum lint insiro novamente

**Pull request merged**